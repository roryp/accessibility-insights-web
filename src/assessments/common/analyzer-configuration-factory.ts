// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Messages } from 'common/messages';
import { RuleAnalyzerConfiguration } from 'injected/analyzers/analyzer';
import { ScannerUtils } from 'injected/scanner-utils';

export class AnalyzerConfigurationFactory {
    public static forScanner(base: Partial<RuleAnalyzerConfiguration>): RuleAnalyzerConfiguration {
        const defaultValues: Partial<RuleAnalyzerConfiguration> = {
            resultProcessor: (scanner: ScannerUtils) => scanner.getAllCompletedInstances,
            telemetryProcessor: telemetryFactory => telemetryFactory.forAssessmentRequirementScan,
            analyzerMessageType: Messages.Assessment.AssessmentScanCompleted,
        };

        return {
            ...defaultValues,
            ...base,
        } as RuleAnalyzerConfiguration;
    }
}
