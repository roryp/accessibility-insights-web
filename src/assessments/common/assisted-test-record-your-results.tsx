// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import * as Markup from '../markup';

export const AssistedTestRecordYourResults = () => (
    <li>
        Record your results:
        <ol>
            <li>
                Select <Markup.Term>Fail</Markup.Term> for any instances that do not meet the
                requirement.
            </li>
            <li>
                Otherwise, select <Markup.Term>Pass</Markup.Term>. Or, after you have marked all
                failures, select <Markup.Term>Pass unmarked instances.</Markup.Term>
            </li>
        </ol>
    </li>
);
