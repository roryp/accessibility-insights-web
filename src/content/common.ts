// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as react from 'react';

import { ContentCreator } from 'views/content/content-page';
import { GuidanceTitle as gt } from 'views/content/guidance-title';
import { link } from './link';

export const create = ContentCreator(link);
export const React = react;
export const GuidanceTitle = gt;
