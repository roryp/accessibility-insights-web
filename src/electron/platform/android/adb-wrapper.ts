// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

export type DeviceInfo = {
    id: string;
    isEmulator: boolean;
    friendlyName: string;
};

export type PackageInfo = {
    versionCode?: number;
    versionName?: string;
};

export interface AdbWrapper {
    getConnectedDevices(): Promise<Array<DeviceInfo>>;
    getPackageInfo(deviceId: string, packageName: string): Promise<PackageInfo>;
    getDumpsysOutput(deviceId: string, serviceToQuery: string): Promise<string>;
    installService(deviceId: string, apkLocation: string): Promise<void>;
    uninstallService(deviceId: string, packageName: string): Promise<void>;
    setTcpForwarding(deviceId: string, localPort: number, devicePort: number): Promise<number>;
    removeTcpForwarding(deviceId: string, devicePort: number): Promise<void>;
}

export interface AdbWrapperFactory {
    createValidatedAdbWrapper(sdkRoot: string): Promise<AdbWrapper>;
}
