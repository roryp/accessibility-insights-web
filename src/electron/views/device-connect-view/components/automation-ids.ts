// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

export const moreInfoLinkAutomationId = 'more-info-link';
export const detectDeviceAutomationId = 'detect-device';
export const tryAgainAutomationId = 'try-again';
export const rescanAutomationId = 'rescan';
export const leftFooterButtonAutomationId = 'android-left-footer-button';
export const rightFooterButtonAutomationId = 'android-right-footer-button';
