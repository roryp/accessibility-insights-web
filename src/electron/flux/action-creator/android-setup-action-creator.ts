// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AndroidSetupActions } from 'electron/flux/action/android-setup-actions';
import { DeviceInfo } from 'electron/platform/android/adb-wrapper';

export class AndroidSetupActionCreator {
    private readonly scope: string = 'AndroidSetupActionCreator';
    constructor(private readonly setupActions: AndroidSetupActions) {}

    public cancel = () => this.setupActions.cancel.invoke();
    public next = () => this.setupActions.next.invoke();
    public rescan = () => this.setupActions.rescan.invoke();
    public saveAdbPath = (newAdbPath: string) => this.setupActions.saveAdbPath.invoke(newAdbPath);
    public setSelectedDevice = (device: DeviceInfo) =>
        this.setupActions.setSelectedDevice.invoke(device);
    public readyToStart = () => this.setupActions.readyToStart.invoke(null, this.scope);
}
