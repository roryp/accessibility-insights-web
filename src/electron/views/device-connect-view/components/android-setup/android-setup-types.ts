// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { LinkComponentType } from 'common/types/link-component-type';
import { UserConfigurationStoreData } from 'common/types/store-data/user-configuration-store';
import { AndroidSetupActionCreator } from 'electron/flux/action-creator/android-setup-action-creator';
import { AndroidSetupStoreData } from 'electron/flux/types/android-setup-store-data';
import { AndroidSetupStepId } from 'electron/platform/android/setup/android-setup-step-id';
import { FolderPickerDeps } from 'electron/views/device-connect-view/components/android-setup/folder-picker';
import * as React from 'react';

export type AndroidSetupStep = React.ComponentType<CommonAndroidSetupStepProps>;

export type CommonAndroidSetupStepProps = {
    userConfigurationStoreData: UserConfigurationStoreData;
    androidSetupStoreData: AndroidSetupStoreData;
    deps: AndroidSetupPageDeps;
};

export type AndroidSetupPageDeps = {
    androidSetupActionCreator: AndroidSetupActionCreator;
    androidSetupStepComponentProvider: AndroidSetupStepComponentProvider;
    LinkComponent: LinkComponentType;
    closeApp: () => void;
    startTesting: () => void;
} & FolderPickerDeps;

export type AndroidSetupStepComponentProvider = Record<AndroidSetupStepId, AndroidSetupStep>;
