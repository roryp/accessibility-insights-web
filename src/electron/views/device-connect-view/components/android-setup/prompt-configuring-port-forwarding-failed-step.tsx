// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import {
    DeviceDescription,
    DeviceDescriptionProps,
} from 'electron/views/device-connect-view/components/android-setup/device-description';
import { tryAgainAutomationId } from 'electron/views/device-connect-view/components/automation-ids';
import { PrimaryButton } from 'office-ui-fabric-react';
import * as React from 'react';
import { AndroidSetupStepLayout, AndroidSetupStepLayoutProps } from './android-setup-step-layout';
import { CommonAndroidSetupStepProps } from './android-setup-types';
import * as styles from './prompt-install-failed-step.scss';

export const PromptConfiguringPortForwardingFailedStep = NamedFC<CommonAndroidSetupStepProps>(
    'PromptConfiguringPortForwardingFailedStep',
    (props: CommonAndroidSetupStepProps) => {
        const { LinkComponent } = props.deps;

        const layoutProps: AndroidSetupStepLayoutProps = {
            headerText: 'Port forwarding failed',
            moreInfoLink: (
                <LinkComponent href="https://aka.ms/accessibility-insights-for-android/portForwarding">
                    Troubleshooting port forwarding problems
                </LinkComponent>
            ),
            leftFooterButtonProps: {
                text: 'Cancel',
                onClick: _ => props.deps.androidSetupActionCreator.cancel(),
            },
            rightFooterButtonProps: {
                text: 'Next',
                disabled: true,
                onClick: undefined,
            },
        };

        const descriptionProps: DeviceDescriptionProps = {
            deviceInfo: props.androidSetupStoreData.selectedDevice,
            className: styles.deviceDescription,
        };

        return (
            <AndroidSetupStepLayout {...layoutProps}>
                <p>
                    We were unable to configure port forwarding from Accessibility Insights for
                    Android Service on your device. Please reconnect your device and try again.
                </p>
                <DeviceDescription {...descriptionProps}></DeviceDescription>
                <PrimaryButton
                    data-automation-id={tryAgainAutomationId}
                    text="Try again"
                    onClick={props.deps.androidSetupActionCreator.next}
                />
            </AndroidSetupStepLayout>
        );
    },
);
