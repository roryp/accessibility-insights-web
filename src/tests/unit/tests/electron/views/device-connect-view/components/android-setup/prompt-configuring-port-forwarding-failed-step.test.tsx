// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AndroidSetupActionCreator } from 'electron/flux/action-creator/android-setup-action-creator';
import { DeviceInfo } from 'electron/platform/android/adb-wrapper';
import { CommonAndroidSetupStepProps } from 'electron/views/device-connect-view/components/android-setup/android-setup-types';
import { PromptConfiguringPortForwardingFailedStep } from 'electron/views/device-connect-view/components/android-setup/prompt-configuring-port-forwarding-failed-step';
import { tryAgainAutomationId } from 'electron/views/device-connect-view/components/automation-ids';
import { shallow } from 'enzyme';
import * as React from 'react';
import { AndroidSetupStepPropsBuilder } from 'tests/unit/common/android-setup-step-props-builder';
import { IMock, Mock, Times } from 'typemoq';

describe('PromptConfiguringPortForwardingFailedStep', () => {
    let props: CommonAndroidSetupStepProps;
    let androidSetupActionCreatorMock: IMock<AndroidSetupActionCreator>;

    beforeEach(() => {
        androidSetupActionCreatorMock = Mock.ofType(AndroidSetupActionCreator);
        props = new AndroidSetupStepPropsBuilder('prompt-configuring-port-forwarding-failed')
            .withDep('androidSetupActionCreator', androidSetupActionCreatorMock.object)
            .build();
    });

    it('renders with device', () => {
        const selectedDevice: DeviceInfo = {
            isEmulator: false,
            friendlyName: 'Super-Duper Gadget',
            id: '1',
        };

        props.androidSetupStoreData.selectedDevice = selectedDevice;

        const rendered = shallow(<PromptConfiguringPortForwardingFailedStep {...props} />);
        expect(rendered.getElement()).toMatchSnapshot();
    });

    it('renders with emulator', () => {
        const selectedDevice: DeviceInfo = {
            isEmulator: true,
            friendlyName: 'Emulator Extraordinaire',
            id: '1',
        };

        props.androidSetupStoreData.selectedDevice = selectedDevice;

        const rendered = shallow(<PromptConfiguringPortForwardingFailedStep {...props} />);
        expect(rendered.getElement()).toMatchSnapshot();
    });

    it('routes try again button to next action', () => {
        const rendered = shallow(<PromptConfiguringPortForwardingFailedStep {...props} />);
        const button = rendered.find({ 'data-automation-id': tryAgainAutomationId });
        button.simulate('click');
        androidSetupActionCreatorMock.verify(m => m.next(), Times.once());
    });
});
