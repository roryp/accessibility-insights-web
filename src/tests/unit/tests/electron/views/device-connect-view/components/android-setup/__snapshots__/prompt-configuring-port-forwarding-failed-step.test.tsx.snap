// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromptConfiguringPortForwardingFailedStep renders with device 1`] = `
<AndroidSetupStepLayout
  headerText="Port forwarding failed"
  leftFooterButtonProps={
    Object {
      "onClick": [Function],
      "text": "Cancel",
    }
  }
  moreInfoLink={
    <LinkComponent
      href="https://aka.ms/accessibility-insights-for-android/portForwarding"
    >
      Troubleshooting port forwarding problems
    </LinkComponent>
  }
  rightFooterButtonProps={
    Object {
      "disabled": true,
      "onClick": undefined,
      "text": "Next",
    }
  }
>
  <p>
    We were unable to configure port forwarding from Accessibility Insights for Android Service on your device. Please reconnect your device and try again.
  </p>
  <DeviceDescription
    className="deviceDescription"
    deviceInfo={
      Object {
        "friendlyName": "Super-Duper Gadget",
        "id": "1",
        "isEmulator": false,
      }
    }
  />
  <CustomizedPrimaryButton
    data-automation-id="try-again"
    onClick={[Function]}
    text="Try again"
  />
</AndroidSetupStepLayout>
`;

exports[`PromptConfiguringPortForwardingFailedStep renders with emulator 1`] = `
<AndroidSetupStepLayout
  headerText="Port forwarding failed"
  leftFooterButtonProps={
    Object {
      "onClick": [Function],
      "text": "Cancel",
    }
  }
  moreInfoLink={
    <LinkComponent
      href="https://aka.ms/accessibility-insights-for-android/portForwarding"
    >
      Troubleshooting port forwarding problems
    </LinkComponent>
  }
  rightFooterButtonProps={
    Object {
      "disabled": true,
      "onClick": undefined,
      "text": "Next",
    }
  }
>
  <p>
    We were unable to configure port forwarding from Accessibility Insights for Android Service on your device. Please reconnect your device and try again.
  </p>
  <DeviceDescription
    className="deviceDescription"
    deviceInfo={
      Object {
        "friendlyName": "Emulator Extraordinaire",
        "id": "1",
        "isEmulator": true,
      }
    }
  />
  <CustomizedPrimaryButton
    data-automation-id="try-again"
    onClick={[Function]}
    text="Try again"
  />
</AndroidSetupStepLayout>
`;
